tests/fastai/test_fastai_model_export.py

```diff
@@ -0,0 +1,259 @@
+from __future__ import print_function
+
+import mock
+import os
+import pytest
+import yaml
+from collections import namedtuple
+
+import numpy as np
+import pandas as pd
+import sklearn.datasets as datasets
+from fastai.tabular import tabular_learner, TabularList
+from fastai.metrics import accuracy
+
+import mlflow.fastai
+import mlflow.utils
+from mlflow import pyfunc
+from mlflow.models import Model
+from mlflow.store.artifact.s3_artifact_repo import S3ArtifactRepository
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.utils.file_utils import TempDir
+from mlflow.utils.model_utils import _get_flavor_configuration
+
+from tests.helper_functions import set_boto_credentials  # pylint: disable=unused-import
+from tests.helper_functions import mock_s3_bucket  # pylint: disable=unused-import
+from tests.helper_functions import score_model_in_sagemaker_docker_container
+
+ModelWithData = namedtuple("ModelWithData", ["model", "inference_dataframe"])
+
+
+@pytest.fixture(scope="session")
+def fastai_model():
+    iris = datasets.load_iris()
+    X = pd.DataFrame(iris.data[:, :2], columns=iris.feature_names[:2])
+    y = pd.Series(iris.target, name='label')
+    data = (TabularList.from_df(pd.concat([X, y], axis=1), cont_names=list(X.columns))
+            .split_by_rand_pct(valid_pct=0.1, seed=42)
+            .label_from_df(cols='label')
+            .databunch())
+    model = tabular_learner(data, metrics=accuracy, layers=[3])
+    model.fit(1)
+    return ModelWithData(model=model, inference_dataframe=X)
+
+
+@pytest.fixture
+def model_path(tmpdir):
+    return os.path.join(str(tmpdir), "model")
+
+
+@pytest.fixture
+def fastai_custom_env(tmpdir):
+    conda_env = os.path.join(str(tmpdir), "conda_env.yml")
+    _mlflow_conda_env(
+        conda_env,
+        additional_pip_deps=["fastai", "pytest"])
+    return conda_env
+
+
+@pytest.mark.large
+def test_model_save_load(fastai_model, model_path):
```

Comment:
Could we update this test to verify that the reloaded model produces the same raw output as the original model, in addition to verifying correctness of the `_FastAiModelWrapper` output? e.g. something like:

```
def test_model_save_load(fastai_model, model_path):
    model = fastai_model.model

    mlflow.fastai.save_model(fastai_learner=model, path=model_path)
    reloaded_model = mlflow.fastai.load_model(model_uri=model_path)
    # Verify reloaded model computes same predictions as original model
    test_data = TabularList.from_df(fastai_model.inference_dataframe)
    model.data.add_test(test_data)
    real_preds, real_target = map(lambda output: output.numpy(), model.get_preds(DatasetType.Test))
    reloaded_model.data.add_test(test_data)
    reloaded_preds, reloaded_target = map(lambda output: output.numpy(), reloaded_model.get_preds(DatasetType.Test))
    np.testing.assert_array_almost_equal(real_preds, reloaded_preds)
    np.testing.assert_array_almost_equal(real_target, reloaded_target)

    # Verify pyfunc produces same output as raw model
    reloaded_pyfunc = pyfunc.load_model(model_uri=model_path)
    pyfunc_predictions = reloaded_pyfunc.predict(fastai_model.inference_dataframe)

    assert len(pyfunc_predictions) == len(real_preds)
    assert set(pyfunc_predictions.columns) == {"predictions", "target"}
    for i, prediction_arr in enumerate(pyfunc_predictions["predictions"]):
        np.testing.assert_array_almost_equal(real_preds[i], prediction_arr)
    for i, target in enumerate(pyfunc_predictions["target"]):
        assert real_target[i] == target
```